### DISCUSSION TITLE
Timestamp

### DISCUSSION THREAD

**Discussion by Lockie85** [blakeblackshear/frigate#7200] [Ask A Question]
Hey,

I'm not aware of a way to overlay a timestamp for camera which don't have the feature built in. I wanted to double check this feature doesn't exist and I'm just missing it?

If it doesn't, perhaps we could get it added as I think it would be super useful. The function (of sorts) is already there as there is an option to add a timestamp within camera debug mode. I expect this has some form of performance hit perhaps. Maybe that could be improved if so, or if not, maybe some people can afford that hit.

Thanks.

**Comment by NickM-27** [comment]
This can be done with ffmpeg already.

But most likely you wouldn't want to, it requires decoding and reencoding the stream which means it will be lower quality and use a lot of CPU / GPU resources.

**Reply by Lockie85** [reply]
Thanks for the reply.

Is this what the debug feature does?

I'd like to give it a go, just to see how much it affects my own context. Is there any directions anywhere I could follow to achieve this?

**Reply by NickM-27** [reply]
The timestamp feature for debug / snapshot simply adds it to the snapshots and decoded frames which is not much work really. If you want it in the recordings then you'll need to use ffmpeg args

**Reply by Lockie85** [reply]
Many thanks
